lib/notifier/sms.js

```diff
@@ -0,0 +1,67 @@
+const _ = require('lodash/fp')
+const utils = require('./utils')
+const sms = require('../sms')
+
+function printSmsAlerts(alertRec, config) {
+  let alerts = []
+
+  if (config.balance) {
+    alerts = _.concat(alerts, alertRec.general)
+  }
+
+  _.keys(alertRec.devices).forEach(function (device) {
+    if (config.balance) {
+      alerts = _.concat(alerts, alertRec.devices[device].balanceAlerts)
+    }
+
+    if (config.errors) {
+      alerts = _.concat(alerts, alertRec.devices[device].deviceAlerts)
+    }
+  })
```

Comment:
Yeah, the same logic is now applied on three different places. Consider the Pull Up Method refactoring

```suggestion
  _.forEach(_.keys(alertRec.devices), (device) => {	
    if (config.balance)
      alerts = _.concat(alerts, alertRec.devices[device].balanceAlerts)

    if (config.errors)
      alerts = _.concat(alerts, alertRec.devices[device].deviceAlerts)
  })
```